daphne_worker/src/config.rs

```diff
@@ -348,10 +345,10 @@ impl DaphneWorkerConfig {
         report_id: &ReportId,
         report_time: Time,
     ) -> String {
-        let mut shard_seed = [0; 8];
-        PrgSha3::seed_stream(&self.report_shard_key, b"report shard", report_id.as_ref())
-            .fill(&mut shard_seed);
-        let shard = u64::from_be_bytes(shard_seed) % self.report_shard_count;
+        let key = ring::hmac::Key::new(ring::hmac::HMAC_SHA256, &self.report_shard_key);
+        let tag = ring::hmac::sign(&key, report_id.as_ref());
+        let shard =
+            u64::from_be_bytes(tag.as_ref()[..8].try_into().unwrap()) % self.report_shard_count;
```

Comment:
use `std::mem::size_of::<u64>()` here instead of `8` so it's really obvious why the number is picked.

Reply:
Done!